### DISCUSSION TITLE
Custom Endpoint Error

### DISCUSSION THREAD

**Discussion by Lucas-Gruber** [directus/directus#7867] [Archive]
Hey, when i register a Custom Endpoint with this Code:
`module.exports = function registerEndpoint(router, { services, exceptions }) {
	const { ItemsService } = services;
	const { ServiceUnavailableException } = exceptions;

	router.get('/', (req, res, next) => {
		const recipeService = new ItemsService('member', { schema: req.schema, accountability: req.accountability });
    const memberService = new ItemsService('member', { schema: req.schema });

		memberService
    	.readByQuery({ sort:'id', fields: ['*'] })
      .then((results) => res.json(results))
      .catch((error) => {
        return next (new ServiceUnavailableException(error.message));
      });
	});
};`

I get only this Response:

`{"errors":[{"message":"query.sort.map is not a function","extensions":{"code":"SERVICE_UNAVAILABLE"}}]}`

**Comment by smilledge** [comment]
In the query, `sort` isn't a string, it's an array of:
```typescript
export declare type Sort = {
        column: string;
        order: 'asc' | 'desc';
};
```

So you probably want something like:
```typescript
memberService.readByQuery({
        sort: [{ column: 'id', order: 'asc' }], 
        fields: ['*']
})
```

**Reply by chrisbartley** [reply]
Ah, thanks to both, I ran into the same problem.  The [custom API endpoint docs](https://docs.directus.io/guides/api-endpoints/?#full-example) need updating, I guess, because the example given uses `readByQuery({ sort: 'name', fields: ['*'] })`

**Reply by rijkvanzanten** [reply]
@chrisbartley Wanna open a PR for those docs? 🙂

**Reply by chrisbartley** [reply]
Done! https://github.com/directus/directus/pull/8088

**Reply by chrisbartley** [reply]
> Ah, thanks to both, I ran into the same problem. The [custom API endpoint docs](https://docs.directus.io/guides/api-endpoints/?#full-example) need updating, I guess, because the example given uses `readByQuery({ sort: 'name', fields: ['*'] })`

New URL for the custom API endpoint recipe docs is: https://docs.directus.io/extensions/endpoints/#example-recipes

**Reply by chrisbartley** [reply]
Note that the API for specifying sort has changed, see https://github.com/directus/directus/discussions/11660#discussioncomment-2184507
